bdikit/mapping_algorithms/column_mapping/algorithms.py

```diff
@@ -102,3 +114,51 @@ def get_column_type(self, context, labels, m=10, model="gpt-4-turbo-preview"):
         )
         col_type_content = col_type.choices[0].message.content
         return col_type_content.split(";")
+
+
+class ContrastiveLearningAlgorithm(BaseColumnMappingAlgorithm):
+    def __init__(self):
+        model_path = get_cached_model_or_download("cl-reducer-v0.1")
+        self.api = ContrastiveLearningAPI(model_path=model_path, top_k=20)
+
+    def map(self, dataset: pd.DataFrame, global_table: pd.DataFrame):
+        union_scopes, scopes_json = self.api.get_recommendations(dataset)
+        matches = {}
+        for column, scope in zip(dataset.columns, scopes_json):
+            candidate = scope["Top k columns"][0][0]
+            if (
+                candidate in global_table.columns
+            ):  # this check protects against the case where the candidate generated from the model is not in the global table
+                matches[column] = candidate
+        return self._fill_missing_matches(dataset, matches)
+
+
+class TwoPhaseMatcherAlgorithm(BaseColumnMappingAlgorithm):
+    def __init__(self):
```

Comment:
To make this matcher more flexible in the future, I would add two parameters to the constructor here: 
```
def __init__(self, model_name: str = "cl-reducer-v0.1", top_k: int = 20):
     ...
```
When we train new models later, we can just instantiate it with different model names.